### PR TITLE
[CELEBORN-1811] Update default value for `celeborn.master.slot.assign.extraSlots`

### PR DESCRIPTION
**Pull request by FMX**
### What changes were proposed in this pull request?
To avoid possible worker load skew for the stages with tiny reducer numbers.

### Why are the changes needed?
If a stage has tiny reducers and skewed partitions, The default value will lead to serious worker load imbalance cause some workers unable to handle shuffle data.

### Does this PR introduce _any_ user-facing change?
Yes

### How was this patch tested?
GA and cluster test.

### CODE REVIEW

**Comment by SteNicholas** [comment]
Please update `migration.md` to describe the change for the default value.

File: docs/configuration/network.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ license: |
...
 | celeborn.&lt;module&gt;.io.lazyFD | true | false | Whether to initialize FileDescriptor lazily or not. If true, file descriptors are created only when data is going to be transferred. This can reduce the number of open files. If setting <module> to `fetch`, it works for worker fetch server. |  |  | 
 | celeborn.&lt;module&gt;.io.maxRetries | 3 | false | Max number of times we will try IO exceptions (such as connection timeouts) per request. If set to 0, we will not do any retries. If setting <module> to `data`, it works for shuffle client push and fetch data. If setting <module> to `replicate`, it works for replicate client of worker replicating data to peer worker. If setting <module> to `push`, it works for Flink shuffle client push data. |  |  | 
-| celeborn.&lt;module&gt;.io.mode | EPOLL | false | Netty EventLoopGroup backend, available options: NIO, EPOLL. If epoll mode is available, the default IO mode is EPOLL; otherwise, the default is NIO. |  |  | 
+| celeborn.&lt;module&gt;.io.mode | NIO | false | Netty EventLoopGroup backend, available options: NIO, EPOLL. If epoll mode is available, the default IO mode is EPOLL; otherwise, the default is NIO. |  |  |
```
Comment (pan3793) [inline docs/configuration/network.md:32]: @SteNicholas could you please change it to optional and explain default behavior in desc?

Comment (SteNicholas) [reply docs/configuration/network.md:32]: @pan3793, +1. I will follow up this config change. @FMX, please revert this change in this pr.

Comment (FMX) [reply docs/configuration/network.md:32]: OK

Comment (FMX) [reply docs/configuration/network.md:32]: If I revert this change, the CI will fail. Just ignore the CI failures related to documentation markdowns.

**Comment by FMX** [comment]
@pan3793 @SteNicholas This PR is ready. How about now?

**Comment by pan3793** [comment]
> Please update `migration.md` to describe the change for the default value.

@FMX same request

**Comment by FMX** [comment]
The migration guide has been updated.

**Review by pan3793 — APPROVED** [review]

**Review by SteNicholas — APPROVED** [review]

**Comment by SteNicholas** [comment]
Merged to main(v0.6.0).

**Comment by s0nskar** [comment]
> To avoid possible worker load skew for the stages with tiny reducer numbers.

@FMX just thinking out loud instead of allocating 100 extra slots for every shuffle running on cluster:
– we can create config "minimum of number of slots" that needs to be allocated for every shuffle.
– extra slots configs can be passed via client, so that such job can be configured appropriately.

**Pull request closed without merging**